flink/flink1.20-shims/src/main/java/org/apache/zeppelin/flink/Flink120Shims.java

```diff
@@ -0,0 +1,423 @@
+    ImmutablePair<Object, Object> pair = createPlannerAndExecutor(
+            classLoader, environmentSettings, senv,
+            tableConfig, moduleManager, functionCatalog, catalogManager);
+    Planner planner = (Planner) pair.left;
+    Executor executor = (Executor) pair.right;
+
+    ResourceManager resourceManager = (ResourceManager) createResourceManager(jars, tableConfig);
+
+    return new StreamTableEnvironmentImpl(catalogManager, moduleManager, resourceManager,
+            functionCatalog, tableConfig, senv, planner, executor, environmentSettings.isStreamingMode());
+  }
+
+  @Override
+  public Object createStreamExecutionEnvironmentFactory(Object streamExecutionEnvironment) {
+    return new StreamExecutionEnvironmentFactory() {
+      @Override
+      public StreamExecutionEnvironment createExecutionEnvironment(Configuration configuration) {
+        return (StreamExecutionEnvironment) streamExecutionEnvironment;
+      }
+    };
+  }
+
+  @Override
+  public Object createCatalogManager(Object config) {
+    final TableConfig tableConfig = TableConfig.getDefault();
+    ClassLoader userClassLoader=Thread.currentThread().getContextClassLoader();
+
+    final CatalogStoreFactory catalogStoreFactory =
+            TableFactoryUtil.findAndCreateCatalogStoreFactory(
+                    tableConfig.getConfiguration(), userClassLoader);
+    final CatalogStore catalogStore =
+                    catalogStoreFactory.createCatalogStore();
+
+    return CatalogManager.newBuilder()
+                    .classLoader(Thread.currentThread().getContextClassLoader())
+                    .config((ReadableConfig) config)
+                    .defaultCatalog(
+                            "default_catalog",
+                            new GenericInMemoryCatalog(
+                                    "default_catalog",
+                                    "default_database"))
+                    . catalogStoreHolder(
+            CatalogStoreHolder.newBuilder()
+                    .classloader(Thread.currentThread().getContextClassLoader())
+                    .config(tableConfig)
+                    .catalogStore(catalogStore)
+                    .factory(catalogStoreFactory)
+                    .build())
+            .build();
+  }
+
+  @Override
+  public String getPyFlinkPythonPath(Properties properties) throws IOException {
+    String mode = properties.getProperty("flink.execution.mode");
+
+    final Path pyDir;
+    if ("yarn-application".equalsIgnoreCase(mode)) {
+      // for yarn-application mode, FLINK_HOME is the container working directory
+      Path flinkHome = Paths.get(".").toAbsolutePath().normalize();
+      pyDir = flinkHome.resolve("lib").resolve("python");
+    } else {
+      String flinkHome = System.getenv("FLINK_HOME");
+      if (flinkHome != null && !flinkHome.isBlank()) {
+        pyDir = Paths.get(flinkHome).resolve("opt").resolve("python");
```

Comment:
Why do you use Paths.resolve() in various places?
I would have shortened it to Paths.get(“flinkHome”, “opt”, "python").

Reply:
fixed.